packages/core/src/application.ts

```diff
@@ -130,7 +147,17 @@ export class Application extends Context {
    * @memberof Application
    */
   public async start(): Promise<void> {
-    await this._forEachServer(s => s.start());
+    debug('Starting the application...');
+    const bindings = this._findLifeCycleObserverBindings();
+    for (const binding of bindings) {
```

Comment:
Have you considered extracting this loop into a function accepting 1) a list of bindings 2) a string name of the method to invoke?

```ts
notifyLifeCycleObservers(bindings, 'start');
```